tests/test_post_gen_project.py

```diff
@@ -0,0 +1,184 @@
+import os
+import shutil
+import subprocess
+
+import pytest
+from conftest import HOME
+
+from hooks.post_gen_project import (  # noqa: E402
+    copy_journal_template_files,
+    get_cookiecutter_dir,
+)
+
+
+@pytest.mark.parametrize(
+    "cookiecutters",
+    [
+        # test get_cookiecutter_dir when "scikit-package-manuscript"
+        #   cookiecutter exists.
+        # C1: with "scikit-package-manuscript" and other cookiecutters,
+        #   expect return the path to "scikit-package-manuscript"
+        ("scikit-package-manuscript", "foo", "bar"),
+    ],
+)
+def test_get_cookiecutter_dir(user_filesystem, cookiecutters):
+    home_path = user_filesystem
+    for cookiecutter in cookiecutters:
+        cookiecutter_path = home_path / ".cookiecutters" / cookiecutter
+        cookiecutter_path.mkdir()
+    actual = get_cookiecutter_dir()
+    expected = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    assert expected == actual
+
+
+@pytest.mark.parametrize(
+    "cookiecutters",
+    [
+        # test get_cookiecutter_dir when "scikit-package-manuscript"
+        #   cookiecutter does not exist.
+        # C1: empty ".cookiecutters", expect FileNotFoundError.
+        (),
+        # C2: without "scikit-package-manuscript", expect FileNotFoundError.
+        ("foo", "bar"),
+    ],
+)
+def test_get_cookiecutter_dir_bad(user_filesystem, cookiecutters):
+    home_path = user_filesystem
+    ck_path = home_path / ".cookiecutters"
+    for cookiecutter in cookiecutters:
+        cookiecutter_path = ck_path / cookiecutter
+        cookiecutter_path.mkdir()
+    with pytest.raises(
+        FileNotFoundError, match=r"Couldn't find scikit-package-manuscript,"
+    ):
+        get_cookiecutter_dir(cookiecutter_name="scikit-package-manuscript")
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test if copy_journal_template_files can find the expected template.
+        # C1: with multiple templates, expect no NotADiectoryError.
+        (["article", "foo", "bar"], "article")
+    ],
+)
+def test_copy_journal_template_files_find_template(
+    user_filesystem, templates, expected_template
+):
+    home_path = user_filesystem
+    skm_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    cwd_path = home_path / "cwd_dir"
+    for template in templates:
+        template_path = skm_path / "templates" / template
+        template_path.mkdir(parents=True, exist_ok=True)
+    copy_journal_template_files(expected_template, cwd_path)
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test copy_journal_template_files when template is not existed.
+        # C1: empty templates, expect NotADirectoryError.
```

Comment:
I don't think we need to test this.  `templates` is not empty (once we successfully found spm)